From eb2f832c69b08786bbbc5f0ce505c0a90ddbae69 Mon Sep 17 00:00:00 2001
From: Valerio Setti <vsetti@baylibre.com>
Date: Wed, 12 Jul 2023 10:30:17 +0200
Subject: [PATCH 38/70] linker: fix RAM placement

New data is taken from Zephyr's "rcar_gen4_cr52.dtsi" file.
A correct placement of RAM data is very important otherwise external
perihperals (such as Rswitch) will not be able to access data structures
placed in RAM.

Signed-off-by: Valerio Setti <vsetti@baylibre.com>
---
 .../linker/gnu_ld/cortex-a/armv8/spider/script.goilTemplate     | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/goil/templates/linker/gnu_ld/cortex-a/armv8/spider/script.goilTemplate b/goil/templates/linker/gnu_ld/cortex-a/armv8/spider/script.goilTemplate
index ebe9b816..ef24f38b 100644
--- a/goil/templates/linker/gnu_ld/cortex-a/armv8/spider/script.goilTemplate
+++ b/goil/templates/linker/gnu_ld/cortex-a/armv8/spider/script.goilTemplate
@@ -6,7 +6,7 @@ ENTRY (_start)
 MEMORY
 {
     FLASH (rx) : ORIGIN = 0x0, LENGTH = (0*1K - 0x0)
-    RAM (wx) : ORIGIN = 0x0, LENGTH = (32512 * 1K - 16 * 1K)
+    RAM (wx) : ORIGIN = 0x40040000, LENGTH = 0x1fc0000
    
     IDT_LIST (wx) : ORIGIN = 0xFFFFF7FF, LENGTH = 2K
 }
-- 
2.34.1

